bin/jsoo_main.ml

```diff
@@ -54,52 +58,52 @@ let implementation ~use_super_errors impl str : Js.Unsafe.obj =
   (* Question ?? *)
   (* let finalenv = ref Env.empty in *)
   let types_signature = ref [] in
-  if use_super_errors then (
-    Misc.Color.setup (Some Always);
-    Lazy.force Super_main.setup);
-
   try
-    Js_config.jsx_version := 3;
     (* default *)
     let ast = impl (Lexing.from_string str) in
-    let ast = Ppx_entry.rewrite_implementation ast in
+    let ast : Parsetree.structure =
+      let ppxlib_ast : Ppxlib_ast__.Versions.OCaml_414.Ast.Parsetree.structure =
+        Obj.magic (ast : Parsetree.structure)
+      in
+      let converted =
+        Convert.copy_structure (Ppxlib.Driver.map_structure ppxlib_ast)
+      in
+      (Obj.magic converted : Parsetree.structure)
```

Comment:
I'm pretty sure this is wrong. You probably copied it from `melc.ml`.

This is applying the PPXes and converting the preprocessed AST to the _current switch_ version. Also `Parsetree.structure` isn't what you think it is because you `open`ed `Melange_compiler_libs` globally.

In the playground, we just need this AST version to be the Melange_compiler_libs parsetree, I think? so we might not need to make the copy at all?

Reply:
I just pushed 53dc2dceba5759a49db71ad00cb59b091e1ba7b9 fixing this

Reply:
> You probably copied it from melc.ml.

Yes.

> In the playground, we just need this AST version to be the Melange_compiler_libs parsetree, I think? so we might not need to make the copy at all?

Just curious, why does ppxlib need all this `copy_structure` calls?